mantidproject/mantid
# Move common functions used in Inelastic/Indirect tabs to separate files

### Description of work
Many common functions implemented in classes of Indirect/Inelastic are either declared within the interface class, which may not be always necessary, or repeated frequently in anonymous namespaces at the beginning of the MVP files. Trying to tidy up the directory, we have moved some of these functions to two different files based on their use: 
 - Functions that manipulate workspaces, typically with a workspace string input, moved to `WorkspaceUtils` 
 - Functions that manipulate few settings of the interface (Such as suffixes) move to `InterfaceUtils`. 

More information in #36610 

Additionally, unit test have been added for these functions. 
#### Summary of work




Fixes #36610 . 




#### Further detail of work


### To test:
- Build Workbench. 
- Inelastic/Indirect interfaces should work as before, with no noticeable change for the user. 






---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
